tree_unique_args/src/body_contains.rs

```diff
@@ -0,0 +1,82 @@
+use crate::ir::{Constructor, ESort, Purpose};
+use strum::IntoEnumIterator;
+
+// Builds rules like:
```

Comment:
nit, optional: comments above functions should have `///` as the prefix and can also have markdown syntax (which helps rustdoc + vscode views of the docs). So this comment could be

```
/// Builds rules like:
/// ```no_run
/// (rule ((Let in out))
///       ((BodyContains (Let in out) out))
///       :ruleset always-run)
/// ```

Reply:
Thanks, updated!